jaraco/windows/clipboard.py

```diff
@@ -82,42 +80,73 @@ class HTMLSnippet(object):
     <https://docs.microsoft.com/en-us/windows/win32/dataxchg/html-clipboard-format>`_.
     """
 
+    # The non-normative grammar defines markers with a space before "-->",
+    # but the body text defines them without spaces and states "with no
+    # whitespace chars within each comment itself".
+    # Actually, Edge and Firefox don't include spaces, so we also don't include
+    # a space.
+    START_FRAGMENT_MARKER = b'<!--StartFragment-->'
```

Comment:
My preference would be to use regular strings throughout the logic (and just account for double-byte characters appropriately). I'm not sure that's possible, but I'd like to explore that possibility. Ideally, the unicode conversions occur as near as possible to the API that's supplying bytes so that the rest of the logic can operate as if the world is bytes and avoid entangling encoding/decoding concerns with other concerns (like `best_type`).
Let me know if you want to explore that possibility; otherwise, I can.